### PR TITLE
[MR-858] Add updateHealthPlanFormData mutation to API

### PR DESCRIPTION
**Pull request by macrael**
## Summary
This is the api side of the work for moving away from the updateDraftSubmission endpoint

#### Related issues
https://qmacbis.atlassian.net/browse/MR-858
https://qmacbis.atlassian.net/browse/MR-856

#### Test cases covered

I added tests for all the errors we can see in the API, I think. 

## QA guidance

I haven’t touched the frontend code yet, I am worried I broke something.

### CODE REVIEW

**Comment by macrael** [comment]
This should be ready for review now. This PR is all API changes, no changes to the FE yet.

File: services/app-api/resolvers/updateHealthPlanFormData.test.ts [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+            throw new Error('type narrow')
+        }
+
+        expect(updateResult.errors[0].extensions?.code).toBe('BAD_USER_INPUT')
+        expect(updateResult.errors[0].message).toContain(
+            'Attempted to update with a StateSubmission'
+        )
+    })
+
+    it('errors if the payload is already submitted', async () => {
```
Comment (haworku) [inline services/app-api/resolvers/updateHealthPlanFormData.test.ts:197]: I _think_ this is a duplicate test. Hmm trying to think.. was it supposed to maybe a test an unlocked and resubmitted submission instead? Or maybe just accidental

**Review by haworku — APPROVED** [review]
Nice work here. I learned a lot reading I like the error handling in the resolvers and the tests

File: services/app-api/resolvers/createSubmission2.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            '037af66b-81eb-4472-8b80-01edf17d12d9',
+        ])
+        expect(draft.documents).toHaveLength(0)
+        expect(draft.managedCareEntities).toHaveLength(0)
+        expect(draft.federalAuthorities).toHaveLength(0)
+        expect(draft.contractDateStart).toBeUndefined()
+        expect(draft.contractDateEnd).toBeUndefined()
+    })
+
+    it('returns an error if the program id is not in valid', async () => {
```
[the remark this answers is not in the document]
Comment (macrael) [reply services/app-api/resolvers/createSubmission2.test.ts:41]: Added the first, I’m not sure about the second b/c we don’t send submission plan data directly, just the three fields: programIDs, submissionType, and submissionDescription.

File: services/app-api/resolvers/updateHealthPlanFormData.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            logError('updateHealthPlanFormData', errMessage)
+            setErrorAttributesOnActiveSpan(errMessage, span)
+            throw new UserInputError(errMessage, {
+                argumentName: 'submissionID',
+            })
+        }
+
+        // Validate input against the db.
+        // Having to crack this open to check on this is probably an indication that some of this info
+        // really belongs on the Submission2 itself instead of being inside form data, but this is where we are now.
```
[the remark this answers is not in the document]
Comment (macrael) [reply services/app-api/resolvers/updateHealthPlanFormData.ts:124]: https://qmacbis.atlassian.net/browse/MR-2093

File: services/app-api/resolvers/updateHealthPlanFormData.test.ts (hunk 1 above)
Comment (macrael) [reply services/app-api/resolvers/updateHealthPlanFormData.test.ts:197]: you are right! but I think I’m missing a test for if the whole submission2 is in a submitted state

File: services/app-api/resolvers/createSubmission2.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (haworku) [reply services/app-api/resolvers/createSubmission2.test.ts:41]: oh right lol!

File: services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx [hunk 4]
```diff
@@ -271,7 +268,7 @@ export const StateSubmissionForm = (): React.ReactElement => {
         )
     }
 
-    if (updateError && !showPageErrorMessage) {
+    if ((updateError || updateFormDataError) && !showPageErrorMessage) {
         setShowPageErrorMessage(true)
```
Comment (haworku) [inline services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx:272]: Can you console error different messages for these two cases similar to the SubmissionSummary page error handling? We want to know (at least from the console and ideally also in logs) when the error is an api request fail versus unwrapping the form data on the client

**Review by haworku — APPROVED** [review]

File: services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx (hunk 4 above)
Comment (macrael) [reply services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx:272]: I’m a little confused about what updateError and updateFormDataError are, actually.

**Pull request merged**